packages/components/tooltip/src/tooltip.tsx

```diff
@@ -1,49 +1,153 @@
+import { Modifiers } from 'popper.js';
+import { SerializedStyles } from '@emotion/utils';
+import type { ElementType, FocusEvent, ChangeEvent, LegacyRef } from 'react';
+
 // inspired from https://github.com/mui-org/material-ui/blob/9ecc8db8abbfb829111d3b5c0678267827984024/packages/material-ui/src/Tooltip/Tooltip.js
 import PropTypes from 'prop-types';
 import React from 'react';
 import { isValidElementType } from 'react-is';
 import isNil from 'lodash/isNil';
 import usePopper from 'use-popper';
 import { useFieldId, useToggleState } from '@commercetools-uikit/hooks';
-import {
-  createSequentialId,
-  SafeHTMLElement,
-  warning,
-} from '@commercetools-uikit/utils';
+import { createSequentialId, warning } from '@commercetools-uikit/utils';
 import { Wrapper, Body, getBodyStyles } from './tooltip.styles';
 
 const sequentialId = createSequentialId('tooltip-');
 
-const TooltipWrapper = (props) => (
+type TComponents = {
+  /**
+   * The component rendered as the tooltip body.
+   */
+  BodyComponent?: ElementType;
+  /**
+   * Where the tooltip should be rendered. This can be useful to render the tooltip into another part of the document, using React portal.
+   */
+  TooltipWrapperComponent?: ElementType;
+  /**
+   * A custom component wrapper for the tooltip.
+   */
+  WrapperComponent?: ElementType;
+};
+
+type TTooltipProps = {
+  children: React.ReactElement;
+
+  /**
+   * Delay (in milliseconds) between the end of the user interaction, and the closing of the tooltip.
+   */
+  closeAfter: number;
+  /**
+   * Custom css-in-js object styles for the tooltip body.
+   */
+  styles: Record<string, SerializedStyles>;
+  /**
+   * Determines if the tooltip should not appear.
+   */
+  off: boolean;
+  /**
+   * An identifier for the tooltip, used for `aria-describedby`.
+   */
+  id?: string;
+  /**
+   * A callback function, called when the tooltip is closing.
+   */
+  onClose?: (e?: ChangeEvent | FocusEvent) => void;
```

Comment:
Suspecting an obvious answer but actually don't know: what makes the event optional? It means every consider has to check for the event being defined.

Reply:
see the implementation block of `handleEnter` and `handleClose` to get an insight on this.
The underlying implementation mixes concerns.

Reply:
Thanks for the reference.

Reply:
Looking at the implementation again, I realised that it wasn't a great idea to mix concerns there.
I considered re-factoring and faking synthetic events.